src/components/DateField/DateFieldInt.part.tsx

```diff
@@ -0,0 +1,367 @@
+import { CustomReactDatepicker } from './CustomReactDatepicker.part';
+import { DatePickerTextField } from './DateFieldTextField.part';
+import { parseDate, safeDateFormat } from '../../utils/date';
+
+export interface DateFieldOpenChangedEvent {
+  /**
+   * If the calendar flyout is supposed to be open or closed.
+   */
+  open: boolean;
+}
+
+const reactDatepickerProps = {
+  adjustDateOnChange: 1,
+  allowSameDay: 1,
+  autoComplete: 1,
+  autoFocus: 1,
+  calendarClassName: 1,
+  calendarContainer: 1,
+  children: 1,
+  className: 1,
+  clearButtonTitle: 1,
+  customInput: 1,
+  customInputRef: 1,
+  dateFormat: 1,
+  dateFormatCalendar: 1,
+  dayClassName: 1,
+  disabled: 1,
+  disabledKeyboardNavigation: 1,
+  dropdownMode: 1,
+  endDate: 1,
+  excludeDates: 1,
+  excludeTimes: 1,
+  filterDate: 1,
+  fixedHeight: 1,
+  forceShowMonthNavigation: 1,
+  formatWeekDay: 1,
+  formatWeekNumber: 1,
+  highlightDates: 1,
+  id: 1,
+  includeDates: 1,
+  includeTimes: 1,
+  injectTimes: 1,
+  inline: 1,
+  isClearable: 1,
+  locale: 1,
+  maxDate: 1,
+  maxTime: 1,
+  minDate: 1,
+  minTime: 1,
+  monthsShown: 1,
+  name: 1,
+  nextMonthButtonLabel: 1,
+  onBlur: 1,
+  onChange: 1,
+  onChangeRaw: 1,
+  onClickOutside: 1,
+  onFocus: 1,
+  onInputClick: 1,
+  onInputError: 1,
+  onKeyDown: 1,
+  onMonthChange: 1,
+  onSelect: 1,
+  onWeekSelect: 1,
+  onYearChange: 1,
+  open: 1,
+  openToDate: 1,
+  peekNextMonth: 1,
+  placeholderText: 1,
+  popperClassName: 1,
+  popperContainer: 1,
+  popperModifiers: 1,
+  popperPlacement: 1,
+  popperProps: 1,
+  preventOpenOnFocus: 1,
+  previousMonthButtonLabel: 1,
+  readOnly: 1,
+  renderCustomHeader: 1,
+  renderDayContents: 1,
+  required: 1,
+  scrollableMonthYearDropdown: 1,
+  scrollableYearDropdown: 1,
+  selected: 1,
+  selectsEnd: 1,
+  selectsStart: 1,
+  shouldCloseOnSelect: 1,
+  showDisabledMonthNavigation: 1,
+  showMonthDropdown: 1,
+  showMonthYearDropdown: 1,
+  showMonthYearPicker: 1,
+  showTimeSelect: 1,
+  showTimeSelectOnly: 1,
+  showWeekNumbers: 1,
+  showYearDropdown: 1,
+  startDate: 1,
+  startOpen: 1,
+  strictParsing: 1,
+  tabIndex: 1,
+  timeCaption: 1,
+  timeFormat: 1,
+  timeIntervals: 1,
+  title: 1,
+  todayButton: 1,
+  useShortMonthInDropdown: 1,
+  useWeekdaysShort: 1,
+  value: 1,
+  weekLabel: 1,
+  withPortal: 1,
+  yearDropdownItemNumber: 1,
+  timeInputLabel: 1,
+  inlineFocusSelectedMonth: 1,
+  onDayMouseEnter: 1,
+  onMonthMouseLeave: 1,
+};
+
+const excludedReactDatePickerProps = {
+  autoComplete: 1,
+  onBlur: 1,
+  onChange: 1,
+  onFocus: 1,
+  children: 1,
+  todayButton: 1,
+  placeholderText: 1,
+  isClearable: 1,
+  withPortal: 1,
+  showYearDropdown: 1,
+  showMonthDropdown: 1,
+  useShortMonthInDropdown: 1,
+  showMonthYearDropdown: 1,
+  dropdownMode: 1,
+  monthsShown: 1,
+  showTimeInput: 1,
+};
+
+interface DatePickerOnChangeEvent extends InputChangeEvent<string> {
+  date: Date;
```

Comment:
why you didn't consider to use `InputChangeEvent<Date>` and pass the date as a `value`?

Reply:
not to introduce breaking changes.